internal/timer/timer.go

```diff
@@ -0,0 +1,13 @@
+package timer
+
+import (
+	"time"
+)
+
+func StopTimer(t *time.Timer) {
```

Comment:
We need to leave a stern comment that this should only be used it the timer hasn't already been drained. If the timer fires, we drain the channel, then we call this function, we'll block forever trying to read from the channel (`t.Stop()` will return `false` because the timer has already fired, `<-t.C` will block because we've already read from the channel).